### ISSUE TITLE
icaltime_as_timet yields incorrect results after year 2100

### ISSUE THREAD

**Issue by minichma** [libical/libical#589]
icaltime_as_timet yields incorrect results for dates/times at or after 2100-03-01.

Consider the following test case, which reproduces the issue (`ok` macro as used in `regresion.c`). The last date/time that yields a correct result is `21000228T235959Z`.

```C
void test_icaltime_as_timet(void) {

    ok("icaltime_from_string translates 19000101T000000Z to -2209075200", icaltime_as_timet(icaltime_from_string("19000101T000000Z")) == -2209075200);
    ok("icaltime_from_string translates 19700101T000000Z to 0", icaltime_as_timet(icaltime_from_string("19700101T000000Z")) == 0);
    ok("icaltime_from_string translates 19700301T235959Z to 5183999", icaltime_as_timet(icaltime_from_string("19700301T235959Z")) == 5183999);
    ok("icaltime_from_string translates 19970717T235959Z to 869183999", icaltime_as_timet(icaltime_from_string("19970717T235959Z")) == 869183999);
    ok("icaltime_from_string translates 20241202T235959Z to 1733183999", icaltime_as_timet(icaltime_from_string("20241202T235959Z")) == 1733183999);
    ok("icaltime_from_string translates 20371231T235959Z to 2145916799", icaltime_as_timet(icaltime_from_string("20371231T235959Z")) == 2145916799);
#if (SIZEOF_TIME_T > 4)
    ok("icaltime_from_string translates 20520419T235959Z to 2597183999", icaltime_as_timet(icaltime_from_string("20520419T235959Z")) == 2597183999);
    ok("icaltime_from_string translates 20790905T235959Z to 3461183999", icaltime_as_timet(icaltime_from_string("20790905T235959Z")) == 3461183999);
    ok("icaltime_from_string translates 21000101T235959Z to 4102531199", icaltime_as_timet(icaltime_from_string("21000101T235959Z")) == 4102531199);
    ok("icaltime_from_string translates 21000228T235959Z to 4107542399", icaltime_as_timet(icaltime_from_string("21000228T235959Z")) == 4107542399);

// Fails from here on
    ok("icaltime_from_string translates 21000301T000000Z to 4107542400", icaltime_as_timet(icaltime_from_string("21000301T000000Z")) == 4107542400);
    ok("icaltime_from_string translates 22370122T235959Z to 8427628799", icaltime_as_timet(icaltime_from_string("22370122T235959Z")) == 8427628799);
    ok("icaltime_from_string translates 23731215T235959Z to 12747628799", icaltime_as_timet(icaltime_from_string("23731215T235959Z")) == 12747628799);
    ok("icaltime_from_string translates 25101107T235959Z to 17067628799", icaltime_as_timet(icaltime_from_string("25101107T235959Z")) == 17067628799);
    ok("icaltime_from_string translates 25821231T000000Z to 19344355200", icaltime_as_timet(icaltime_from_string("25821231T000000Z")) == 19344355200);
#endif
}
```

**Comment by minichma** [comment]
This seems to be related to the following code, which doesn't handle the missing leap day in 2100: https://github.com/libical/libical/blob/b8c2f4cfdf940687e96d1433493012cd8cd0b2e3/src/libical/icaltime.c#L143-L147

**Comment by winterz** [comment]
fixed with 6e4dab56cf1383b4d2650be019199eedf4f22755
